src/executor/JobExecutor.ts

```diff
@@ -23,38 +23,43 @@ export class JobExecutor {
   }
 
   async execute(jobEntity: JobEntity, parameters?: JobParameters): Promise<JobResult> {
-    const { added, running } = await this.schedulesRepository.addExecution(jobEntity.name, jobEntity.maxRunning);
-    if (!added) {
-      this.logger.debug('maxRunning reached, skip', {
-        name: jobEntity.name,
-        running,
-      });
-      return {
-        status: ExecutionStatus.maxRunningReached,
-      };
-    }
+    try {
```

Comment:
why do you wrap it in try-catch here? We are catching potential errors in the method calling this one. Doesn´t catching the error here completely break the error handling we are currently trying to implement?

Reply:
we definitely miss an integration test